lambda/script.py

```diff
@@ -121,6 +125,9 @@ def get_group_info(
             )
         except HttpError as e:
             print(f"Http error for group {group_id}: {e}")
+            if "exceeded" in f"{e}".lower():
+                print("Rate limit exceeded, stopping loop over groups...")
```

Comment:
are we likely to exceed our rate limit? If so, we're just exiting out -  does this mean we're losing data?

Reply:
We could be, and in fact, we probably are... I just had a look at the CloudWatch logs and since this change the lambda for the Cabinet Office domain has been emitting a new error:
`"Quota exceeded for quota metric 'Queries' and limit 'Queries per minute per user' of service 'groupssettings.googleapis.com' for consumer ...`

Previously, we'd get an error for hitting the quota for 'Queries per day', which meant that for several hours of each day every lambda invocation would fail until that quota would reset. Now because we've parallelised the lambdas, we seem to hit a *new* limit (the per minute request quota), and are logging the Google Group data we've collected "so far".

I believe that if we fix this 'Queries per minute' issue, we'll be hitting the 'Queries per day' error again. Honestly, that would be preferable, because right now we're getting *incomplete* information on a regular basis, while we used to get *complete* (I think) information, just less frequently than we wanted (about once or twice a day instead of every hour).

Until the quota limits are raised, I don't think we'll be getting the data we want (which is a description of how *all* the Google Groups in a domain are configured at the start of every hour). I think we need to talk to several people:
- Those that are expecting the Google Groups data being in Splunk on time and in full, to warn them of potential gaps in the data
- Those that can request a quota limit increase from Google on the CO domain. We can probably figure out what both the 'Queries per minute' limit should be, given an API call takes about 180ms, and the 'Queries per day' limit, as we know how many Groups there are in the domain and that we're making a request for each one 24 times a day.